src/Features/Core/Portable/CodeCleanup/DiagnosticSet.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CodeCleanup
+{
+    internal class DiagnosticSet
```

Comment:
EnabledDiagnosticOptions should probably renamed to CodeCleanUpOptions or something

basically, EnabledDiagnosticOption is union of all options affecting fixers used in code cleanup feature. if we can change those 2 organizeusing service to analyzer and fixer then we probably need list of DiagnosticSet

diagnostic set just group multiple diagnostic id to 1 unit of code cleanup fixer.

...

we probably need to decide whether we will expose the fact that we use code fixer as code cleanup and expose it (description and etc) to new editor code clean up feature as it is from code fixer (the same thing we expose on ruleset editor). if we choose this way, I believe we need to use origanize using and formatting from code fixer rather than service directly and we can expose options as they are right now. 

or we can create new code cleaner concept which has its own description/options and etc more suitable for code cleanup. and we expose new code cleaner to new editor code cleanup platform.

the code cleaner can internally use code fixer but that will become an implementation detail. and it would be fine a code cleaner to not use code fixer (such as formatting or organize using service. or even code refactoring such as sync folder type name and etc). one of problem of such approach will be 2 are quite similar thing packaged in different way. meaing just more boiler code. but we could add some helper there which we expose code fix as code cleaner to editor so reduce some of the boiler code writing required from users.